src/box/iproto.cc

```diff
@@ -2080,6 +2097,10 @@ iproto_on_accept(struct evio_service *service, int fd,
 	 */
 	msg = iproto_msg_new(con);
 	if (msg == NULL) {
+		/* con->shutdown_msg may be used in shutdown process. */
+		if (rlist_empty(&con->in_all_list))
```

Comment:
Here is memory leak! In case this if is true, below code is not executed!

Reply:
`if` is `true` only if shutdown started (else `iproto_thread->all_connections` also in the same list) and after `shutdown_timeout` seconds memory will freed.